internal/mage/sdk/nodejs.go

```diff
@@ -88,13 +89,54 @@ func (t NodeJS) Generate(ctx context.Context) error {
 	})
 }
 
-// Publish publishes the NodeJS SDK
-func (t NodeJS) Publish(ctx context.Context, tag string) error {
-	panic("FIXME")
+// Publish publishes the Node.js SDK
+func (t Nodejs) Publish(ctx context.Context, tag string) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	var (
+		version     = strings.TrimPrefix(tag, "sdk/nodejs/v")
+		tokenSecret = c.Host().EnvVariable("NPM_TOKEN").Secret()
+	)
+	token, err := tokenSecret.Plaintext(ctx)
+	if err != nil {
+		return err
+	}
+
+	if token == "" {
+		return errors.New("NPM_TOKEN environment variable must be set")
+	}
+
+	build := nodeJsBase(c).Exec(dagger.ContainerExecOpts{
+		Args: []string{"npm", "run", "build"},
+	})
+
+	// configure .npmrc
+	npmrc := fmt.Sprintf(`//registry.npmjs.org/:_authToken=%s
+registry=https://registry.npmjs.org/
+always-auth=true`, token)
+	if err := os.WriteFile("sdk/nodejs/.npmrc", []byte(npmrc), 0600); err != nil {
```

Comment:
nit: This should be done in container rather than on the host machine

Reply:
A worthy improvement for a follow-up. Leaving this tab open post-release.